Adds `compatible` key check to gameSupported;

agc93 opened the pull request:
This is the result of Nexus-Mods/Vortex#9665 ;

So this is the simplest method I could think of: a game could add a simple `dinput: false` to the `compatible` object and the dinput mod type would regard it as not supported entirely.

As always, open to suggestions, and if this is an acceptable solution, I can open a matching PR for the ENB mod type following the same convention.

@IDCs (for visibility)

agc93 committed:
Adds `compatible` key check to gameSupported;
Allows for game to specify it doesn't support the mod type;

TanninOne commented:
In my head it would be the same, more generic, compatible flag for enb and dinput, something like "compatible.deployToGameDirectory", or can either of you @agc93 @IDCs think of a situation where one of the two types would work but not the other?

Also the condition could can be slightly simplified: "if (game.compatible?. dinput === false)"
The ? makes it so that if compatible is undefined, compatible?.dinput will also be undefined instead of throwing an exception. Best language feature ever...

IDCs commented:
I think in 99.99% of cases if one of the two works, so should the other one. The only exception to this would be due to the fact that dinput could potentially act as a replacement mod if a game is shipped with a dinput.dll file, while ENB should theoretically (at least as far as I know) always be independent of any game files.

Oddly in some cases (thank you Microsoft for Game Pass) adding new files to a certain directory can potentially work just fine, while attempts at replacing existing files will fail with permission errors. I would say lets be safe rather than sorry and have them as separate properties in the compatible element. `game.compatible?.dinput` and `game.compatible?.enb`

agc93 committed:
Change to support more generic `compatible` flag;
Supports either `deployToGameDirectory` or `dinput`;
Simplified conditional syntax;

agc93 commented:
Thanks for the clarification @TanninOne I couldn't remember when null-chaining was introduced so didn't want to use it in case it wasn't available, but I have updated to use that.

I've pre-emptively changed to use either `deployToGameDirectory` or `dinput`, but if you think sticking to one or the other is better, I can update.

_Personally_ I can't see a situation where one is used but not the other, but I also don't use either type frequently enough for that to carry much weight.

TanninOne commented:
I can't see why we'd need to disable one of the modtypes and not the other but also it doesn't hurt to have both options.
Thank you for the contribution!
